app/src/main/java/ru/alexzdns/fundamentals/homework/ui/moviesList/MoviesListViewModelFactory.kt

```diff
@@ -0,0 +1,13 @@
+package ru.alexzdns.fundamentals.homework.ui.moviesList
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class MoviesListViewModelFactory : ViewModelProvider.Factory {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T = when (modelClass) {
+        MoviesListViewModel::class.java -> MoviesListViewModel()
```

Comment:
Фабрика - подходящее место для подготовки зависимостей.
```suggestion
        MoviesListViewModel::class.java -> MoviesListViewModel(context = App.context())
```